File: documentation/user/en/query/filtering/examples/hierarchy/hierarchy-within-reference-having.cs
EvitaResponse<SealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => session.QuerySealedEntity(
        Query(
        	Collection("Product"),
        	FilterBy(
        		HierarchyWithin(
        			"categories",
        			AttributeEquals("code", "accessories"),
        			Having(
        				Or(
        					AttributeIs("validity", Null),
        					AttributeInRange(
        						"validity", 
        						DateTimeOffset.Parse("2023-10-01T01:00:00-01:00", DateTimeFormatInfo.InvariantInfo)
        					)
        				)
        			)
        		)
        	),
        	Require(
        		EntityFetch(
        			AttributeContent("code")
        		)
        	)
        )
	)
);

File: documentation/user/en/query/filtering/examples/hierarchy/hierarchy-within-reference-having.java
final EvitaResponse<SealedEntity> entities = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				collection("Product"),
				filterBy(
					hierarchyWithin(
						"categories",
						attributeEquals("code", "accessories"),
						having(
							or(
								attributeIs("validity", NULL),
								attributeInRange(
									"validity", 
									OffsetDateTime.parse("2023-10-01T01:00:00-01:00", DateTimeFormatter.ISO_OFFSET_DATE_TIME)
								)
							)
						)
					)
				),
				require(
					entityFetch(
						attributeContent("code")
					)
				)
			)
		);
	}
);